jgrapht-core/src/main/java/org/jgrapht/graph/AsSynchronizedGraph.java

```diff
@@ -0,0 +1,1114 @@
+         */
+        @Override
+        public Iterator<E> iterator()
+        {
+            return getCopy().iterator();
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public Object[] toArray()
+        {
+            synchronized (mutex) {
+                return set.toArray();
+            }
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public <T> T[] toArray(T[] a)
+        {
+            synchronized (mutex) {
+                return set.toArray(a);
+            }
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public boolean add(E e)
+        {
+            throw new UnsupportedOperationException("the Set is unmodifiable");
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public boolean remove(Object o)
+        {
+            throw new UnsupportedOperationException("the Set is unmodifiable");
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public boolean containsAll(Collection<?> c)
+        {
+            synchronized (mutex) {
+                return set.containsAll(c);
+            }
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public boolean addAll(Collection<? extends E> c)
+        {
+            throw new UnsupportedOperationException("the Set is unmodifiable");
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public boolean retainAll(Collection<?> c)
+        {
+            throw new UnsupportedOperationException("the Set is unmodifiable");
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public boolean removeAll(Collection<?> c)
+        {
+            throw new UnsupportedOperationException("the Set is unmodifiable");
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public void clear()
+        {
+            throw new UnsupportedOperationException("the Set is unmodifiable");
+        }
+
+
+        /**
+         * {@inheritDoc}
+         */
+        // Override default methods in Collection
+        @Override
+        public void forEach(Consumer<? super E> action)
+        {
+            synchronized (mutex) {
+                set.forEach(action);
+            }
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public boolean removeIf(Predicate<? super E> filter)
+        {
+            throw new UnsupportedOperationException("the Set is unmodifiable");
+        }
+
+        /**
+         * Creates a <Code>Spliterator</code> over the elements in the set's copy which is unmodifiable.
+         *
+         * @return a  <code>Spliterator</code> over the elements in the backing set's copy which is
+         * unmodifiable.
+         */
+        @SuppressWarnings("unchecked")
```

Comment:
Why are these SuppressWarnings required?

Reply:
Sorry, I am not familiar with these methods. I added SuppressWarnings just because unmodifiableCollection adds it.

Reply:
If it's not necessary, remove it.